src/System.Net.NameResolution/src/System/Net/DNS.cs

```diff
@@ -358,16 +445,119 @@ private static IPHostEntry HostResolutionEndHelper(IAsyncResult asyncResult)
             return (IPHostEntry)castedResult.Result;
         }
 
-        private static IAsyncResult BeginGetHostEntry(string hostNameOrAddress, AsyncCallback requestCallback, object stateObject)
+        [Obsolete("BeginGetHostByName is obsoleted for this type, please use BeginGetHostEntry instead. http://go.microsoft.com/fwlink/?linkid=14202")]
+        public static IAsyncResult BeginGetHostByName(string hostName, AsyncCallback requestCallback, object stateObject)
+        {
+            if (NetEventSource.Log.IsEnabled()) NetEventSource.Enter(NetEventSource.ComponentType.Socket, "DNS", "BeginGetHostByName", hostName);
```

Comment:
Nit: I don't know if we've done this in other places, but these strings could use nameof as well, e.g. nameof(DNS) and nameof(BeginGetHostByName)